sdv/single_table/base.py

```diff
@@ -1335,14 +1336,29 @@ def _sample_with_conditions(
 
             condition = dict(zip(condition_columns, group))
             condition_df = dataframe.iloc[0].to_frame().T
-            try:
-                transformed_condition = self._data_processor.transform(
-                    condition_df, is_condition=True
-                )
-            except ConstraintsNotMetError as error:
-                raise ConstraintsNotMetError(
-                    'Provided conditions are not valid for the given constraints.'
-                ) from error
+            if hasattr(self, '_chained_patterns'):
+                try:
+                    transformed_condition = self._transform_helper(condition_df)
```

Comment:
just marking that this will have collisions with this PR https://github.com/sdv-dev/SDV/pull/2480/files